pages/api/handler/links.js

```diff
@@ -0,0 +1,119 @@
+import Link from "../../../models/Link";
+import Profile from "../../../models/Profile";
+import Stats from "../../../models/Stats";
+import connectMongo from "../../../config/mongo";
+
+/**
+ * @type {import('next').NextApiHandler}
+ */
+export default async function handler(req, res) {
+  await connectMongo();
+
+  const { username, url } = req.query;
+
+  if (req.method != "GET" || !username || !url) {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: username and link required" });
+  }
+
+  let getProfile = await Profile.findOne({ username });
+
+  if (!getProfile) {
+    try {
+      await Profile.create({
+        username,
+        views: 1,
+      });
+
+      getProfile = await Profile.findOne({ username });
+    } catch (e) {
+      const err = `ERROR creating profile: ${e}`;
+      console.log(err);
+      return res.status(500).json({ message: err });
+    }
+  }
+
+  const getLink = await Link.findOne({ username, url });
+
+  if (getLink) {
+    try {
+      await Link.updateOne(
+        {
+          username,
+          url,
+        },
+        {
+          $inc: { clicks: 1 },
+        }
+      );
+    } catch (e) {
+      const err = `ERROR incrementing link stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  if (!getLink) {
+    try {
+      const link = await Link.create({
+        profile: getProfile._id,
+        username,
+        url,
+        clicks: 1,
+      });
+
+      await Profile.updateOne(
+        {
+          username,
+        },
+        {
+          $push: { links: link._id },
+        },
+        { new: true, useFindAndModify: false }
+      );
+    } catch (e) {
+      const err = `ERROR creating link: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  const date = new Date();
+
+  date.setHours(1, 0, 0, 0);
+
+  const getPlatformStats = await Stats.findOne({ date });
+
+  if (getPlatformStats) {
+    try {
+      await Stats.updateOne(
+        {
+          date,
+        },
+        {
+          $inc: { clicks: 1 },
+        }
+      );
+    } catch (e) {
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  if (!getPlatformStats) {
+    try {
+      await Stats.create({
+        date,
+        clicks: 1,
+      });
+    } catch (e) {
+      const err = `ERROR incrementing platform stats: ${e}`;
+      console.log(err);
+      return res.res(500).json({ message: err });
+    }
+  }
+
+  return res.status(201).redirect(decodeURIComponent(url));
```

Comment:
This file looks very similar to the existing api handler file `pages/api/statistics/[...data].js`, we don't need both especially now as the other one doesn't seem to be used anymore.

I prefer the other one because it handles the uri as part of the path rather than optional query parameters.

Could we add this redirect line to the existing file `pages/api/statistics/[...data].js`? https://github.com/EddieHubCommunity/LinkFree/blob/main/pages/api/statistics/%5B...data%5D.js

Or am I missing something?

Reply:
Yes, but that will be naming things badly, because now it does not just handle the stats, but also the link, the process of the link. Also the query parameters aren't optional, you give the query parameters you get an answer, if you don't, you get a error. I prefer this approach because if a person goes to just api/statistics they get a 404 code, or they go to just api/statistics/:username, the still just get a 404 code. Parameters will come useful incontext to the coming API 
![image](https://user-images.githubusercontent.com/91655303/212467072-30b73bea-0132-4b26-a68f-8494c85027d6.png)

Reply:
I know you have checked for those parameters, but from what I remember for REST, query parameters are usually for optional ones (regardless if you check for them or not), where when it is part of the path, it is for the resource

For example a filter would be a query parameter (like keyword or sort) but the resource of what is being filter (like users or posts) would be part of the path.

Reply:
Maybe it shouldn't be on `statistics`, but I still think it should be part of the path, similar to how it is now

I think the best option is to add the redirect to the existing file, we don't need unused code, and move that file out of `statistics`

Reply:
So you mean deleting the file statistics and making it a part of the path? 👍

Reply:
Probably better to use that file and just move it instead of deleting it